shared/src/test/scala/mlscript/DiffTests.scala

```diff
@@ -354,6 +367,16 @@ class DiffTests extends org.scalatest.funsuite.AnyFunSuite with org.scalatest.Pa
               }
             )
             
+            if (!varianceWarnings.isEmpty) {
+              import Message._
+              var warnings: List[Message -> Opt[Loc]] = Nil
+              varianceWarnings.foreach{ case (tname, biVars) =>
+                warnings ::= (msg"${tname} does not use ${biVars.mkString(",")}" -> None)
```

Comment:
> ${tname} does not use ${biVars.mkString(",")}

It may actually be using it these. For example in `class Foo[A] method F: Foo[A]`. So I'd rephrase this as I suggested in the issue, "type parameter T is irrelevant and could be removed".

Also, better generate one warning line per type parameter, this way each one will get its own location.

```suggestion
                warnings :::= biVars.map(v => msg"type parameter $v is irrelevant and could be removed" -> ...)
```